[DOC] Update docs for 24.02.0 release [skip ci] (#10387)
NVIDIA/spark-rapids: docs/download.md

### RAPIDS Accelerator's Support Policy for Apache Spark
The RAPIDS Accelerator maintains support for Apache Spark versions available for download from [Apache Spark](https://spark.apache.org/downloads.html)

### Download RAPIDS Accelerator for Apache Spark v24.02.0
| Processor | Scala Version | Download Jar | Download Signature |
|-----------|---------------|--------------|--------------------|
| x86_64    | Scala 2.12    | [RAPIDS Accelerator 2.12 v24.02.0](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/24.02.0/rapids-4-spark_2.12-24.02.0.jar) | [Signature](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/24.02.0/rapids-4-spark_2.12-24.02.0.jar.asc) |
| x86_64    | Scala 2.13    | [RAPIDS Accelerator 2.13 v24.02.0](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.13/24.02.0/rapids-4-spark_2.13-24.02.0.jar) | [Signature](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.13/24.02.0/rapids-4-spark_2.13-24.02.0.jar.asc) |
| arm64     | Scala 2.12    | [RAPIDS Accelerator 2.12 v24.02.0](#) | [Signature](#) |

> | arm64     | Scala 2.13    | [RAPIDS Accelerator 2.13 v24.02.0](#) | [Signature](#) |

Reviewer: These links need to be fixed

Author: Yes, I'll keep that in mind. I don't know what these links are going to be as it's the first release. I'll update them in a different PR when we update gh-pages.

Reviewer: Check with @NvTimLiu, these should be flled out here, not hacked into gh-pages afterwards unless we're for some reason not releasing ARM binaries simultaneously?

Author: Sure!

Reviewer 2: I think the arm jars will be include the cuda classifier, and will be named 

rapids-4-spark_2.12-24.04.0-cuda11-arm64.jar
rapids-4-spark_2.12-24.04.0-cuda12-arm64.jar

rapids-4-spark_2.13-24.04.0-cuda11-arm64.jar
rapids-4-spark_2.13-24.04.0-cuda12-arm64.jar

Related issue: https://github.com/NVIDIA/spark-rapids/issues/6881

Author: Drafted [10414](https://github.com/NVIDIA/spark-rapids/pull/10414) to fix this, thanks!

> | arm64     | Scala 2.12    | [RAPIDS Accelerator 2.12 v24.02.0](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/24.02.0/rapids-4-spark_2.12-24.02.0-arm64.jar) | [Signature](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/24.02.0/rapids-4-spark_2.12-24.02.0-arm64.jar.asc) |

Reviewer 3: @SurajAralihalli  As Sameer mentioned : https://github.com/NVIDIA/spark-rapids/pull/10387#discussion_r1484988693

The arm jar names are combined with CUDA classifiers,  

As this PR got merged, can you help follow it up? Thanks!

Author: Drafted [10414](https://github.com/NVIDIA/spark-rapids/pull/10414) to fix this, thanks!

### RAPIDS Accelerator's Support Policy for Apache Spark
The RAPIDS Accelerator maintains support for Apache Spark versions available for download from [Apache Spark](https://spark.apache.org/downloads.html)

### Download RAPIDS Accelerator for Apache Spark v24.02.0

> - **x86**

Reviewer: ```suggestion
- **x86_64**
```

	- **Scala 2.12:**
	- [RAPIDS Accelerator for Apache Spark 24.02.0 - Scala 2.12 jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/24.02.0/rapids-4-spark_2.12-24.02.0.jar)
	- [RAPIDS Accelerator for Apache Spark 24.02.0 - Scala 2.12 jar.asc](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/24.02.0/rapids-4-spark_2.12-24.02.0.jar.asc)

	- **Scala 2.13:**
	- [RAPIDS Accelerator for Apache Spark 24.02.0 - Scala 2.13 jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.13/24.02.0/rapids-4-spark_2.13-24.02.0.jar)
	- [RAPIDS Accelerator for Apache Spark 24.02.0 - Scala 2.13 jar.asc](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.13/24.02.0/rapids-4-spark_2.13-24.02.0.jar.asc)

> - **arm64**

Author: @jlowe what do you think of adding arm64 jars this way?

Reviewer: This layout is fine, could also do a table layout with processor, Scala version, jar link, and signature link as columns.

* Download the [PUB_KEY](https://keys.openpgp.org/search?q=sw-spark@nvidia.com).
* Import the public key: `gpg --import PUB_KEY`
* Verify the signature for Scala 2.12 jar:
    `gpg --verify rapids-4-spark_2.12-24.02.0.jar.asc rapids-4-spark_2.12-24.02.0.jar`
* Verify the signature for Scala 2.13 jar:
    `gpg --verify rapids-4-spark_2.13-24.02.0.jar.asc rapids-4-spark_2.13-24.02.0.jar`

The output of signature verify:

	gpg: Good signature from "NVIDIA Spark (For the signature of spark-rapids release jars) <sw-spark@nvidia.com>"

### Release Notes
New functionality and performance improvements for this release include:

> * Discontinued support for Nvidia GPUs based on Pascal architecture.

Author: @viadea @nvliyuan

Please provide suggestions for 24.02 release notes

Reviewer 4: Added more reviewers to comment on what we should highlight in release notes for 24.02.

* Download the [PUB_KEY](https://keys.openpgp.org/search?q=sw-spark@nvidia.com).
* Import the public key: `gpg --import PUB_KEY`
* Verify the signature for Scala 2.12 jar:
    `gpg --verify rapids-4-spark_2.12-24.02.0.jar.asc rapids-4-spark_2.12-24.02.0.jar`
* Verify the signature for Scala 2.13 jar:
    `gpg --verify rapids-4-spark_2.13-24.02.0.jar.asc rapids-4-spark_2.13-24.02.0.jar`

The output of signature verify:

	gpg: Good signature from "NVIDIA Spark (For the signature of spark-rapids release jars) <sw-spark@nvidia.com>"

### Release Notes
New functionality and performance improvements for this release include:
* Discontinued support for Nvidia GPUs based on Pascal architecture.
* Set get_json_object functionality to disabled by default.
* Implemented string comparison in AST expressions.
* Expanded timezone support to include options beyond UTC.

> * Enhanced security by adding checksum for cached files in Filecache.

Reviewer: Checksums are disabled by default, and it's not really a security feature.  If an attacker can overwrite the cached data they can also overwrite the checksum.
```suggestion
* Optional checksums for cached files in the file cache.
```

* Introduced support for Databricks 13.3 ML LTS.

> * Added support for parse_url functionality.

Reviewer 4: parse_url support for PATH, QUERY, PROTOCOL.

Reviewer 4: Support for Lazy Quantifier for regular expression functions

Reviewer 4: Support for format_number function

Reviewer 4: Batching support for row-based bounded window functions

> * Qualification and Profiling tool:

Reviewer: The qualification and profiling tools are released separately and on an independent schedule.  Does it make sense to list things here vs. point to the tools download page and let it describe what the recent releases are and what they have?

Reviewer 4: How about referring to https://github.com/NVIDIA/spark-rapids-tools/releases @mattahrens @amahussein  ?
